libsolidity/inlineasm/AsmCodeGen.cpp

```diff
@@ -214,9 +215,29 @@ class CodeTransform: public boost::static_visitor<>
 	{
 		size_t numVariables = m_state.variables.size();
 		std::for_each(_block.statements.begin(), _block.statements.end(), boost::apply_visitor(*this));
-		// pop variables
-		// we deliberately do not check stack height
+
 		m_state.assembly.setSourceLocation(_block.location);
+
+		// issue warnings for stack height discrepancies
+		int deposit = m_state.assembly.deposit();
```

Comment:
I think you have to check that `m_state.assembly.deposit()` is not changed by line 216.